controllers/usersignupcleanup/usersignup_cleanup_controller.go

```diff
@@ -108,9 +110,9 @@ func (r *Reconciler) Reconcile(request reconcile.Request) (reconcile.Result, err
 
 	// Check if the UserSignup is:
 	// * either deactivated
-	// * or if it is waiting for phone verification to be finished and at the same time the status contains the approved condition set to true
+	// * or if it is waiting for phone verification to be finished and at the same it has at least one activation stored in the annotation
 	//   (in other words the UserSignup is reactivated but the phone verification hasn't been finished)
-	if states.Deactivated(instance) || (states.VerificationRequired(instance) && condition.IsTrue(instance.Status.Conditions, toolchainv1alpha1.UserSignupApproved)) {
+	if states.Deactivated(instance) || (states.VerificationRequired(instance) && activationsAnnotationPresent && activations != "0") {
```

Comment:
if a user is already deactivated, her UserSignup resource should always have the `activation-counter` annotation (with a value >= `1`)

Reply:
However, at this point, the `UserSignup` CR is not in the deactivated state anymore - that's why we check if the state contains `verificationRequired` and then also the annotation.

Reply:
still, if the user has been deactivated in the past, doesn't it imply that she already has the `activation-counter` annotation with value equal or greater than 1? 
(btw, I find `states.Deactivated(...)` to be a bit confusing: at first sight, I assume that this func checks the *current* state)

Reply:
in this part, we want to clean:
* deactivated UserSignups that has been deactivated for more than one year
* UserSignups that were deactivated, the user tried to reactivated but hasn't verified the phone for more than one year

that being said - we need to check both:
* if the current state is `deactivated`
* or the current state is `verificationRequired` and the number of activations is > 0

Reply:
ok, got it now. Please ignore my previous comments, they were wrong 🤦‍♂️